app/models/result.rb

```diff
@@ -4,8 +4,7 @@ class Result < ApplicationRecord
 
   before_save :denormalize_department
 
-  belongs_to :assessment, counter_cache: true
-  belongs_to :assignment, required: true
+  belongs_to :assignment
```

Comment:
`required: true` adds a model validation. In this commit, we change to use a database constrain instead. A model validation isn't particularly useful here because if the user somehow manages to submit a request to create a result that doesn't have an associated assignment, then we've probably got a bug somewhere and a validation error message wouldn't help.

I wrote an article about this not too long ago: https://robots.thoughtbot.com/validation-database-constraint-or-both